script.js

```diff
@@ -1,66 +1,151 @@
 const DONE_TASK_TEXT_CLASS = "task-input-text-corssed";
 
-function addEmptyTask() {
-    let task = createEmptyTask();
-    content.appendChild(task);
-}
+class Task{
+    constructor(){
+        this.div = document.createElement("div");
+        this.div.classList.add("task");
+        this.div.task = this;
+        this.taskInputpreviousText = "";
+        this.taskInput = null;
+        this.deleteButton = null;
+        this.finishedButton = null;
+        this.addTaskInputBar()
+        this.addTaskDeleteButton()
+        this.addTaskFinishedButton()
 
-function createRawTask() {
-    // this function creates just the task object with the relevent event handlers, without inner elements
-    let newTask = document.createElement("div");
-    newTask.classList.add("task")
-    return newTask;
-}
+    }
+
+    getState(){
+        return {
+            text: this.taskInputpreviousText,
+            done: this.isDone(),
+        };
+    }
 
-function addTaskInputBar(task) {
-    // this function will add to the task the input bar section, and will return the input bar object.
-    let taskInput = document.createElement("input");
-    taskInput.classList.add("task-input-text");
-    taskInput.addEventListener("keydown", (event) => {
-        if (event.key === "Enter") {
-            taskInput.blur();
+    setState(state){
+        this.taskInputpreviousText = state.text;
+        this.taskInput.value = this.taskInputpreviousText;
+        if (state.done){
+            this.markDone();
         }
-    })
-    task.appendChild(taskInput);
-    return taskInput;
+    }
+
+    handleKeydownEvent(event){
+        this.text = this.taskInput.value;
+        switch(event.key){
+            case "Enter":
+                this.taskInputpreviousText = this.taskInput.value;
+                this.taskInput.blur();
+                break;
+
+            case "Escape":
+                this.taskInput.blur();         
+        }
+    }
+
+    addTaskInputBar() {
+        this.taskInput = document.createElement("input");
+        this.taskInput.classList.add("task-input-text");
+        this.taskInput.addEventListener("keydown", (event) => {this.handleKeydownEvent(event)})
+        this.taskInput.addEventListener("blur", (event) => {this.taskInput.value = this.taskInputpreviousText})
+        this.div.appendChild(this.taskInput);
+    }
+
+    addTaskDeleteButton() {
+        this.deleteButton = document.createElement("button");
+        this.deleteButton.textContent = "X";
+        this.deleteButton.classList.add("delete-button");
+        this.deleteButton.onclick = () => {this.deleteTask();};
+        this.div.appendChild(this.deleteButton);
+    }
+
+    addTaskFinishedButton() {
+        this.finishedButton = document.createElement("button");
+        this.finishedButton.textContent = '\u{02713}';
+        this.finishedButton.classList.add("finished-button");
+        this.finishedButton.onclick = () => { this.changeTaskDoneMark(this.taskInput) };
```

Comment:
do it without arrow func. 
hint: read about `.bind`, `.apply` and `.call`

Reply:
refactored the code, so this part is irrelevant